qiskit_nature/problems/second_quantization/lattice/models/heisenberg_model.py

```diff
@@ -0,0 +1,131 @@
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Heisenberg model."""
+
+import numpy as np
+import logging
+from typing import Optional, Tuple
+from fractions import Fraction
+from qiskit_nature.second_q.operators import SpinOp
+from qiskit_nature.problems.second_quantization.lattice.lattices import Lattice
+from .lattice_model import LatticeModel
+
+logger = logging.getLogger(__name__)
+
+
+class HeisenbergModel(LatticeModel):
+    """The Heisenberg model."""
+
+    def __init__(
+        self,
+        lattice: Lattice,
+        coupling_constants: Tuple = (1.0, 1.0, 1.0),
+        ext_magnetic_field: Tuple = (0.0, 0.0, 0.0),
+    ) -> None:
+        """
+        Args:
+            lattice: Lattice on which the model is defined.
+            coupling_constants (Tuple, optional): Coupling constants. Defaults to (1.0, 1.0, 1.0).
+            ext_magnetic_field (Tuple, optional): External magnetic field. Defaults to (0.0, 0.0, 0.0).
+        """
+        super().__init__(lattice)
+        self.coupling_constants = coupling_constants
+        self.ext_magnetic_field = ext_magnetic_field
+
+    def coupling_matrix(self) -> np.ndarray:
+        """Return the coupling matrix."""
+        return self.interaction_matrix()
+
+    @classmethod
+    def uniform_parameters(
+        cls, lattice: Lattice, uniform_interaction: complex, uniform_onsite_potential: complex
+    ) -> "HeisenbergModel":
+        """Set a uniform interaction parameter and on-site potential over the input lattice.
+
+        Args:
+            lattice (Lattice): Lattice on which the model is defined.
+            uniform_interaction (complex): The interaction parameter.
+            uniform_onsite_potential (complex): The on-site potential.
+
+        Returns:
+            HeisenbergModel: The Heisenberg model with uniform parameters.
+        """
+        return cls(
+            cls._generate_lattice_from_uniform_parameters(
+                lattice, uniform_interaction, uniform_onsite_potential
+            )
+        )
+
+    @classmethod
+    def from_parameters(cls, interaction_matrix: np.ndarray) -> "HeisenbergModel":
+        """Return the Hamiltonian of the Heisenberg model
+        from the given interaction matrix and on-site interaction.
+
+        Args:
+            interaction_matrix (np.ndarray): A real or complex valued squared matrix.
+
+        Returns:
+            HeisenbergModel: The Heisenberg model generated from the given interaction
+                matrix and on-site interaction.
+        """
+        return cls(cls._generate_lattice_from_parameters(interaction_matrix))
+
+    def second_q_ops(
+        self,
+        display_format: Optional[str] = None,
```

Comment:
You can remove this argument. It will be removed from the other classes, too :+1: